weather_mv/loader_pipeline/metrics.py

```diff
@@ -90,50 +93,77 @@ def wrapper(self, *args, **kwargs):
                 raise ValueError("Function is not a generator.")
 
         return wrapper
+
     return decorator
 
 
 class AddTimer(beam.DoFn):
-    """DoFn to add a empty time_dict per element in PCollection. This dict will stage_names as keys
-    and the time it took for that element in that stage."""
+    """DoFn to add a empty time_dict per element in PCollection. This dict will
+    stage_names as keys and the time it took for that element in that stage."""
+
     def process(self, element) -> t.Iterator[t.Any]:
-        time_dict = {}
+        time_dict = OrderedDict([
+            ("uri", element),
+            ("bucket", get_file_time(element)),
```

Comment:
nit: we should rename these as well `bucket` & `pickup`.

Reply:
I kept it like this so it's more readable in the logs. i.e.

```
... Time from FileInit -> bucket: 5039 seconds.
... Time from bucket -> pickup: 1494498 seconds.
...Time from pickup -> FilterFileTransform: 3 seconds.
...
```